### PR TITLE
Previous / next project buffer commands

### LINKED ISSUE #896: Cycle through current project's buffer's only
Is there a way to cycle through buffers like with next-buffer/previous-buffer but limited only to buffers associated with the current project?  If not, this would be great -- perhaps projectile could add [buffer ring](http://www.emacswiki.org/emacs/BufferRing) integration to achieve this?

### PR DESCRIPTION
**Pull request by mookid**
Add commands to switch to previous / next buffers in the current project.

- [x] update menu
- [x] update manual
- [x] handle non-project case

Fix #896.

### CODE REVIEW

File: projectile.el [hunk 1]
```diff
@@ -4018,6 +4019,8 @@ thing shown in the mode line otherwise."
     (define-key map (kbd "x t") #'projectile-run-term)
     (define-key map (kbd "x s") #'projectile-run-shell)
     (define-key map (kbd "z") #'projectile-cache-current-file)
+    (define-key map (kbd "<left>") #'projectile-previous-project-buffer)
```
Comment (bbatsov) [inline projectile.el]: Those should also be mentioned in the menu and the manual.

File: projectile.el [hunk 2]
```diff
@@ -4137,6 +4140,32 @@ Otherwise behave as if called interactively.
...
+    (when (cdr-safe projectile-project-buffers)
+      (while (and (< counter max-iterations)
+                  (not (gethash (current-buffer) other-project-buffers)))
+        (apply orig-fun args)
+        (incf counter)))))
+
+(defun projectile-next-project-buffer ()
+  "In selected window switch to the next project buffer."
+  (interactive)
+  (projectile--repeat-until-project-buffer #'next-buffer))
```
Comment (bbatsov) [inline projectile.el]: How does the command behave when invoked outside a project? I guess it should just show some error as suggesting to switch to a project in this case doesn't make much sense.

Comment (mookid) [reply projectile.el]: alternatively, it may (warn and) forward to previous-buffer / next-buffer

Comment (bbatsov) [reply projectile.el]: Yeah, that might actually be even better.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**